feat: Agregar sistema de registro de usuarios

Uriel020 opened the pull request:
- Crear páginas de registro paso 1 y paso 2 con validación completa
- Implementar navegación entre pasos con React Router
- Agregar estilos CSS personalizados con diseño moderno y responsive
- Integrar formularios con validación en tiempo real
- Persistir datos entre pasos usando localStorage
- Añadir enlace de registro en el header
- Configurar rutas para /registro y /registro/paso2 

#36

4brm01 committed:
feat: Agregar sistema de registro de usuarios

- Crear páginas de registro paso 1 y paso 2 con validación completa
- Implementar navegación entre pasos con React Router
- Agregar estilos CSS personalizados con diseño moderno y responsive
- Integrar formularios con validación en tiempo real
- Persistir datos entre pasos usando localStorage
- Añadir enlace de registro en el header
- Configurar rutas para /registro y /registro/paso2

jaime-hndz reviewed (dismissed):
piola

Albert2707 reviewed (changes requested):
No guardar la contraseña en el localStorage o data sensible dentro de este, si necesitas urgentemente guardar esta data, usa sessionStorage.

4brm01 committed:
Refactor: Implementar nuevo diseño y mejoras de seguridad

- Se aplicó el nuevo diseño
- Se quitó el console.log
- Ya no se usa alert y se cambió por sweetalert2

Uriel020 reviewed (commented):
@4brm01 

Revisar todo lo que está en español, menos los comentarios que vayan al usuario pasarlos a inglés. Igual los nombres del archivo.

Las páginas registro paso 1 y paso 2, son 1 sola página llamada registro con 2 componentes que cambian (aunque se podría hacer perfectamente con 1 solo componente).

La rama tiene conflictos, debes hacer pull a los cambios y revisar que no choque con lo que tienes
